### PR TITLE
Support architecture parameter

### PR DESCRIPTION
**Pull request by dcvz**
#### Changes

- Provides `--architecture` parameter for support on running on M1 agents

#### Related Issues

- TBD

#### Checklist

- [x] Read the contribution [guide](https://github.com/game-ci/unity-builder/blob/main/CONTRIBUTING.md) and accept the
      [code](https://github.com/game-ci/unity-builder/blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Docs (If new inputs or outputs have been added or changes to behavior that should be documented. Please make a PR
      in the [documentation repo](https://github.com/game-ci/documentation))
- [x] Readme (updated or not needed)
- [ ] Tests (added, updated or not needed)

### CODE REVIEW

**Review by webbertakken — APPROVED** [review]
Nice one!

File: src/model/platform-setup/setup-mac.ts [hunk 1]
```diff
@@ -72,6 +72,26 @@ class SetupMac {
     return '';
   }
 
+  private static getArchitectureParameters(): string[] {
+    const architectureArgument = [];
+
+    // @ts-ignore
+    const process = require('process');
```
Comment (webbertakken) [inline src/model/platform-setup/setup-mac.ts:79]: I'm not sure but I don't think you need to import process and use TS ignore. Node has process as a global.

Comment (dcvz) [reply src/model/platform-setup/setup-mac.ts:79]: Oh, alright! I'll try with the change

**Comment by dcvz** [comment]
Looks like CI builds is failing because of the license server

File: src/model/platform-setup/setup-mac.ts (hunk 1 above)
Comment (dcvz) [reply src/model/platform-setup/setup-mac.ts]: Changed made, and it works too!

**Comment by AndrewKahr** [comment]
Hey there, we just merged a fix for the failing license activation. Can you merge main into your branch and link a successful build log from your own branch. Then we can review and merge. Thanks!

**Comment by dcvz** [comment]
Can take a look at this today! Or is this already part of #596?

**Comment by AndrewKahr** [comment]
I've already added it to the other PR and merged it in an effort to get v4 out faster. I will make sure you are credited in the release notes and commit message though for your contribution! Gonna close this PR now since the functionality is now merged to main.

**Pull request closed without merging**